synapse/handlers/register.py

```diff
@@ -149,7 +162,7 @@ def register(
                     user_id = None
                     token = None
                     attempts += 1
-                    if attempts > 5:
+                    if attempts > 50:
```

Comment:
We're now deterministically generating the things, so there's more likelihood of collisions.

On matrix.org we're ok with attempts > 2, but other servers may have more than 5 sequentially allocated numeric IDs